File: SampleDemo/app/src/main/java/com/adform/adformdemo/DemoApplication.java
package com.adform.adformdemo;

import android.app.Application;

import com.adform.sdk.utils.AdApplicationService;

/**
 * An extended application class, that extends Adform SDK functionality.
 * Note: This class should be defined in AndroidManifest.xml to work properly.
 */
public class DemoApplication extends Application implements AdApplicationService.ServiceListener {

    private AdApplicationService adService;

    @Override
    public void onCreate() {
        super.onCreate();
        // [mandatory] Initializes application service
        adService = AdApplicationService.init();
    }

    // [mandatory] A mandatory method for the SDK to work properly
    @Override
    public AdApplicationService getAdService() {
        return adService;
    }
}

File: SampleDemo/app/src/main/java/com/adform/adformdemo/DemoApplication.kt
package com.adform.adformdemo

import androidx.multidex.MultiDexApplication
import com.adform.sdk.utils.AdApplicationService

/**
 * An extended application class, that extends Adform SDK functionality.
 * Note: This class should be defined in AndroidManifest.xml to work properly.
 */
class DemoApplication : MultiDexApplication(), AdApplicationService.ServiceListener {

    private lateinit var adService: AdApplicationService

    override fun onCreate() {
        super.onCreate()
        // [mandatory] Initializes application service
        adService = AdApplicationService.init()
    }

    // [mandatory] A mandatory method for the SDK to work properly
    override fun getAdService(): AdApplicationService {
        return adService
    }
}
